cmd/sonictool/check/live.go

```diff
@@ -44,7 +45,7 @@ func checkLiveBlockRoot(dataDir string, cacheRatio cachescale.Func) error {
 	if lastBlock == nil {
 		return fmt.Errorf("verification failed - unable to get the last block (%d) from gdb", lastBlockIdx)
 	}
-	err = gdb.EvmStore().CheckLiveStateHash(lastBlockIdx, lastBlock.Root)
+	err = gdb.EvmStore().CheckLiveStateHash(lastBlockIdx, hash.Hash(lastBlock.StateRoot))
```

Comment:
future work: should we try to unify hash types ?

Reply:
Very future work, yes. Likely not in this codebase.